### PR TITLE
Caching headers

### PR DESCRIPTION
**Pull request by JeanMertz**
* disable caching of `503 Service Unavailable` responses
* disable caching of all API resources by default (can be overwritten)

closes https://github.com/blendle/eipiai/issues/20

### CODE REVIEW

File: lib/eipiai/resources/base.rb [hunk 1]
```diff
@@ -75,9 +75,23 @@ def query_keys
...
+    # When the service is unavailable, the `Cache-Control` header is set to not
+    # allow any caching by proxies/clients, and sets the `Retry-After` header to
+    # tell the client to retry again in 60 seconds.
+    #
+    # @return [true, false]
+    #
     def service_available?
       return true if Eipiai::HealthCheck.new.healthy?
 
+      response.headers['Cache-Control'] = 'no-cache, no-store, must-revalidate, private'
```
Comment (jurriaan) [inline lib/eipiai/resources/base.rb]: Why don't you use `cache_header` in here

Comment (jurriaan) [inline lib/eipiai/resources/base.rb]: Nevermind, it's weird to do so

Comment (JeanMertz) [inline lib/eipiai/resources/base.rb]: Yeah, `cache_header` can be overwritten, but this one should be explicitly overwritten if you want to change this.

File: lib/eipiai/resources/base.rb [hunk 2]
```diff
@@ -225,6 +243,16 @@ def create_uri
 
     private
 
+    # cache_header
+    #
+    # The default Cache-Control header value served by all resources.
+    #
+    # @return String default Cache-Control header value
+    #
+    def cache_header
```
Comment (JeanMertz) [inline lib/eipiai/resources/base.rb]: I should probably rename this to something more descriptive, like `cache_control_header_value`.

**Pull request merged**